src/middlewared/middlewared/api/v25_04_0/acl.py

```diff
@@ -0,0 +1,372 @@
+            raise ValueError(
+                'At least one of FILE_INHERIT or DIRECTORY_INHERIT must '
+                'be set if INHERIT_ONLY is present in the ACE flags'
+            )
+
+        return self
+
+
+class NFS4ACE_BasicFlags(BaseModel):
+    BASIC: NFS4ACE_BasicFlagset
+
+
+class NFS4ACE(BaseModel):
+    tag: NFS4ACE_Tags
+    type: NFS4ACE_EntryTypes
+    perms: NFS4ACE_AdvancedPerms | NFS4ACE_BasicPerms
+    flags: NFS4ACE_AdvancedFlags | NFS4ACE_BasicFlags
+    id: AceWhoId | None = None
+    who: LocalUsername | RemoteUsername | None = None
+
+    @model_validator(mode='after')
+    def check_ace_valid(self) -> Self:
+        if self.tag in NFS4_SPECIAL_ENTRIES:
+            if self.id not in (-1, None):
+                raise ValueError(
+                    f'{self.id}: id may not be specified for the following ACL entry '
+                    f'tags: {", ".join([tag for tag in NFS4_SPECIAL_ENTRIES])}'
+                )
+        else:
+            if not self.who and self.id in (None, -1):
+                raise ValueError(
+                    'Numeric ID "id" or account name "who" must be specified'
+                )
+
+        return self
+
+
+class NFS4ACL_Flags(BaseModel):
+    autoinherit: bool = False
+    protected: bool = False
+    defaulted: bool = False
+
+
+POSIXACE_Tags = Literal[
+    POSIXACE_Tag.USER_OBJ,
+    POSIXACE_Tag.GROUP_OBJ,
+    POSIXACE_Tag.OTHER,
+    POSIXACE_Tag.MASK,
+    POSIXACE_Tag.USER,
+    POSIXACE_Tag.GROUP
+]
+
+
+class POSIXACE_Perms(BaseModel):
+    READ: bool
+    WRITE: bool
+    EXECUTE: bool
+
+
+class POSIXACE(BaseModel):
+    tag: POSIXACE_Tags
+    perms: POSIXACE_Perms
+    default: bool
+    id: AceWhoId | None = None
+    who: LocalUsername | RemoteUsername | None = None
+
+    @model_validator(mode='after')
+    def check_ace_valid(self) -> Self:
+        if self.tag in POSIX_SPECIAL_ENTRIES:
+            if self.id not in (-1, None):
+                raise ValueError(
+                    f'{self.id}: id may not be specified for the following ACL entry '
+                    f'tags: {", ".join([tag for tag in POSIX_SPECIAL_ENTRIES])}'
+                )
+        else:
+            if not self.who and self.id in (None, -1):
+                raise ValueError(
+                    'Numeric ID "id" or account name "who" must be specified'
+                )
+
+        return self
+
+
+class AclBaseInfo(BaseModel):
+    uid: AceWhoId | None
+    gid: AceWhoId | None
+
+
+class NFS4ACL(AclBaseInfo):
+    acltype: Literal[FS_ACL_Type.NFS4]
+    acl: list[NFS4ACE]
+    aclflags: NFS4ACL_Flags
+    trivial: bool
+
+
+class POSIXACL(AclBaseInfo):
+    acltype: Literal[FS_ACL_Type.POSIX1E]
+    acl: list[POSIXACE]
+    trivial: bool
+
+
+class DISABLED_ACL(AclBaseInfo):
+    # ACL response paths with ACL entirely disabled
+    acltype: Literal[FS_ACL_Type.DISABLED]
+    acl: Literal[None]
+    trivial: Literal[True]
+
+
+class FilesystemGetaclArgs(BaseModel):
```

Comment:
This is real picky, but my concern is that we won't be able to change the model names in future versions without breaking backwards compatibility. Could we consider `FilesystemGetAclArgs` or `FilesystemGetACLArgs`?

(same thing for Result and setacl)

Reply:
Okay. I was matching the method name `filesystem.getacl` and `filesystem.setacl`.